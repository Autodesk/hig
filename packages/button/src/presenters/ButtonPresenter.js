import React, { Component } from "react";
import PropTypes from "prop-types";
import { cx, css } from "emotion";
import { ThemeContext } from "@hig/theme-context";

import stylesheet from "./stylesheet";

import {
  availableTargets,
  availableTypes,
  availableWidths
} from "../constants";

export default class ButtonPresenter extends Component {
  static propTypes = {
    disabled: PropTypes.bool,
    hasFocus: PropTypes.bool,
    hasHover: PropTypes.bool,
    isPressed: PropTypes.bool,
    icon: PropTypes.node,
    link: PropTypes.string,
    onBlur: PropTypes.func,
    onClick: PropTypes.func,
    onFocus: PropTypes.func,
    onHover: PropTypes.func,
    onMouseDown: PropTypes.func,
    onMouseEnter: PropTypes.func,
    onMouseLeave: PropTypes.func,
    onMouseUp: PropTypes.func,
    stylesheet: PropTypes.func,
    target: PropTypes.oneOf(availableTargets),
    title: PropTypes.string.isRequired,
    type: PropTypes.oneOf(availableTypes),
    width: PropTypes.oneOf(availableWidths)
  };

  render() {
    const {
      disabled,
      hasFocus,
      hasHover,
      isPressed,
      icon,
      link,
      onBlur,
      onClick,
      onFocus,
      onHover,
      onMouseDown,
      onMouseEnter,
      onMouseLeave,
      onMouseUp,
      stylesheet: customStylesheet,
      target,
      title,
      type,
      width,
      ...otherProps
    } = this.props;

    const { className } = otherProps;

    const href = link || undefined;
    const tabIndex = disabled ? "-1" : "0";
    const Wrapper = link ? "a" : "button";
    const wrapperTarget = link ? target : undefined;

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles, metadata }) => {
          const styles = stylesheet(
            { disabled, hasFocus, hasHover, isPressed, type, width },
            resolvedRoles,
            metadata
          );
          const cssStyles = customStylesheet
            ? customStylesheet(styles, this.props, resolvedRoles, metadata)
            : styles;

          return (
            <Wrapper
              {...otherProps}
              className={cx(css(cssStyles.button), className)}
              href={href}
              tabIndex={tabIndex}
              target={wrapperTarget}
              onBlur={onBlur}
              onClick={onClick}
              onFocus={onFocus}
              onMouseDown={onMouseDown}
              onMouseEnter={onMouseEnter}
              onMouseLeave={onMouseLeave}
              onMouseOver={onHover}
              onMouseUp={onMouseUp}
              disabled={disabled}
            >
              {icon && <span className={css(cssStyles.icon)}>{icon}</span>}
              <span className={icon ? css(cssStyles.iconText) : ""}>
                {title}
              </span>
            </Wrapper>
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
