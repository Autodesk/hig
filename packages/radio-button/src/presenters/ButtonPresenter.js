import React, { Component } from "react";
import PropTypes from "prop-types";
import { ThemeContext } from "@hig/theme-context";
import { cx, css } from "emotion";
import { createCustomClassNames } from "@hig/utils";

import stylesheet from "./stylesheet";

export default class ButtonPresenter extends Component {
  static propTypes = {
    /**
     * Adds custom/overriding styles
     */
    stylesheet: PropTypes.func
  };

  render() {
    const { stylesheet: customStylesheet, ...otherProps } = this.props;
    const { className } = otherProps;
    const spanClassName = createCustomClassNames(className, "span");

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles, metadata }) => {
          const styles = stylesheet(
            {
              stylesheet: customStylesheet,
              ...this.props
            },
            resolvedRoles,
            metadata.colorSchemeId
          );
          return (
            <span
              className={cx(css(styles.radioButtonWrapper), spanClassName)}
            />
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
