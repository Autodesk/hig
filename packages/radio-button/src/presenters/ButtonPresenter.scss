@import '~@hig/styles/mixins/colors';
@import '~@hig/styles/tokens/layers';

.hig__radio-button__wrapper {
  position: relative;
  color: transparent;
  border: 1px solid color(hig-cool-gray-30);
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 3px;

  .hig__radio-button__input:hover + & {
    border-color: color(hig-turquoise-50);
  }

  .hig__radio-button__input[disabled] + & {
    border-color: color(hig-cool-gray-30);
    border-style: dotted;
    background-color: transparent;
    pointer-events: none;
  }

  .hig__radio-button__input:focus + & {
    outline: none;
    border-color: color(hig-blue-60);
  }

  .hig__icon{
    position: initial;
  }

  .hig__radio-button & {
    border-radius: 50%;
  }

  .hig__radio-button .hig__radio-button__input:focus:checked + & {
    border-color: color(hig-blue-60);
    background-color: color(hig-blue-60);
    box-shadow: inset 0 0 0 3px color(hig-white);
  }

  .hig__radio-button .hig__radio-button__input:checked + & {
    background: none;
    border-color: color(hig-slate-40);
    background-color: color(hig-slate-40);
    box-shadow: inset 0 0 0 3px color(hig-white);
  }

  .hig__radio-button .hig__radio-button__input:checked[disabled] + & {
    background: none;
    background-color: color(hig-cool-gray-30);
    border-color: color(hig-cool-gray-30);
    border-style: dotted;
  }

  .hig__radio-button .hig__radio-button__input[disabled] + & {
    background: none;
    border-color: color(hig-cool-gray-30);
    border-style: dotted;
  }
}