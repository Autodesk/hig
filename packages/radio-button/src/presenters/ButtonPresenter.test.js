import React from "react";
import renderer from "react-test-renderer";
import ButtonPresenter from "./ButtonPresenter";

describe("radio-button/presenters/ButtonPresenter", () => {
  [
    {
      description: "renders without props",
      props: {}
    },
    {
      description: "renders with all props",
      props: {
        checked: true,
        disabled: true
      }
    }
  ].forEach(({ description, props: { children, ...otherProps } }) => {
    it(description, () => {
      const presenter = (
        <ButtonPresenter {...otherProps}>{children}</ButtonPresenter>
      );
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
