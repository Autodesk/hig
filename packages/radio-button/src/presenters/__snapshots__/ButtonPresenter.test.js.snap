// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio-button/presenters/ButtonPresenter renders with all props 1`] = `
.emotion-0 {
  opacity: 0.4;
  position: relative;
  color: transparent;
  border: none;
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 50%;
  box-shadow: inset 0 0 0 1px  rgba(128,128,128,0.5) ,0 0 0 0px transparent;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
}

<span
  className="emotion-0"
/>
`;

exports[`radio-button/presenters/ButtonPresenter renders without props 1`] = `
.emotion-0 {
  opacity: 1.0;
  position: relative;
  color: transparent;
  border: none;
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 50%;
  box-shadow: inset 0 0 0 1px  rgba(128,128,128,0.5) ,0 0 0 0px transparent;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
}

<span
  className="emotion-0"
/>
`;
